### ISSUE TITLE
Tests test the wrong simulation method

### ISSUE THREAD

**Issue by yaelbh** [Qiskit/qiskit-aer#1021]
# What is the current behavior?
See the lines: https://github.com/Qiskit/qiskit-aer/blob/93cef6cf524bf78250415726d905ba729a3531c4/test/terra/backends/qasm_simulator/qasm_standard_gates.py#L111 and https://github.com/Qiskit/qiskit-aer/blob/93cef6cf524bf78250415726d905ba729a3531c4/test/terra/backends/qasm_simulator/qasm_standard_gates.py#L155

These lines remove the method from the backend options. From now on, all the tests run only the automatic simulation method. This affects:
1. https://github.com/Qiskit/qiskit-aer/blob/master/test/terra/backends/qasm_simulator/qasm_standard_gates.py tests a few dozens of gates. Only the first of them is tested with the designated simulation method. All other gates are tested with the automatic simulation method.
2. Other tests, which reside in different Python files, and are executed after the tests of https://github.com/Qiskit/qiskit-aer/blob/master/test/terra/backends/qasm_simulator/qasm_standard_gates.py. For example, https://github.com/Qiskit/qiskit-aer/blob/master/test/terra/backends/qasm_simulator/qasm_unitary_gate.py is always executed only with the automatic simulation method.

### Steps to reproduce the problem
Add a debug print
```
print(backend._options)
```
in https://github.com/Qiskit/qiskit-aer/blob/master/test/terra/backends/qasm_simulator/qasm_standard_gates.py and other tests.

Also compare the fluent execution of
```
python test/terra/backends/test_qasm_simulator_matrix_product_state.py
```
with the never-ending execution of a specific test:
```
python test/terra/backends/test_qasm_simulator_matrix_product_state.py TestQasmMatrixProductStateSimulator.test_unitary_gate
```

Only in the second execution the test runs with the MPS simulator.

### Suggested solutions

I've added `.copy()` at the end of the problematic lines, but now that I run the tests, three tests are failing (these tests should have failed before, but passed since they were wrongly executed with the automatic simulation method), and eventually get stuck in `test_unitary_gate`. I'm fixing them now.

**Comment by yaelbh** [comment]
Opened #1022 about `test_unitary_gate`, which runs forever.

**Comment by yaelbh** [comment]
Fixed by #1026
